src/trace_redaction/prune_perf_events_integrationtest.cc

```diff
@@ -0,0 +1,133 @@
+namespace {
+constexpr auto kTrace = "test/data/trace-redaction-perf-sample.pftrace";
+
+constexpr auto kPackageName = "com.example.sampleapp";
+constexpr auto kPid = 25131;
+}  // namespace
+
+class PrunePerfEventsIntegrationTest
+    : public testing::Test,
+      protected TraceRedactionIntegrationFixure {
+ protected:
+  void SetUp() override {
+    SetSourceTrace(kTrace);
+
+    trace_processor::Config tp_config_unredacted;
+    trace_processor_unredacted_ =
+        trace_processor::TraceProcessor::CreateInstance(tp_config_unredacted);
+    trace_processor::Config tp_config_redacted;
+    trace_processor_redacted_ =
+        trace_processor::TraceProcessor::CreateInstance(tp_config_redacted);
+
+    TraceRedactor::Config tr_config;
+    auto trace_redactor = TraceRedactor::CreateInstance(tr_config);
+
+    Context context;
+    context.package_name = kPackageName;
+    base::Status status = Redact(*trace_redactor, &context);
+    if (!status.ok()) {
+      PERFETTO_ELOG("Redaction error: %s", status.c_message());
+    }
+    ASSERT_OK(status);
+
+    auto raw = LoadRedacted();
+    ASSERT_OK(raw);
+
+    auto read_buffer = std::make_unique<uint8_t[]>(raw->size());
+    memcpy(read_buffer.get(), raw->data(), raw->size());
+
+    ASSERT_OK(
+        trace_processor_redacted_->Parse(std::move(read_buffer), raw->size()));
+    ASSERT_OK(trace_processor_redacted_->NotifyEndOfFile());
+
+    status = LoadTrace(GetSourceTrace(), trace_processor_unredacted_.get());
+    ASSERT_OK(status);
+  }
+
+  std::unique_ptr<trace_processor::TraceProcessor> trace_processor_unredacted_;
+  std::unique_ptr<trace_processor::TraceProcessor> trace_processor_redacted_;
+};
+
+TEST_F(PrunePerfEventsIntegrationTest, OnlyKeepsTargetProcessPerfSamples) {
+  // This query retrieves the total number of perf samples for target process
+  // in redacted trace
+  auto query =
+      " SELECT COUNT(*) FROM perf_sample "
+      "JOIN thread ON thread.utid = perf_sample.utid "
+      "JOIN process ON process.upid = thread.upid "
+      "GROUP BY pid "
+      "HAVING pid = " +
+      std::to_string(kPid);
+
+  auto rows = trace_processor_redacted_->ExecuteQuery(query);
+  ASSERT_TRUE(rows.Next());
+  int64_t perf_samples_for_target_pid = rows.Get(0).AsLong();
+  ASSERT_TRUE(perf_samples_for_target_pid > 0);
+
+  // This query retrieves the total number of perf samples for all processes
+  // in redacted trace
+  query =
+      " SELECT COUNT(*) FROM perf_sample "
+      "JOIN thread ON thread.utid = perf_sample.utid "
+      "JOIN process ON process.upid = thread.upid";
+  rows = trace_processor_redacted_->ExecuteQuery(query);
+  ASSERT_TRUE(rows.Next());
+  int64_t trace_perf_samples = rows.Get(0).AsLong();
+
+  // Check that all the trace samples in the trace correspond to target process.
+  ASSERT_TRUE(perf_samples_for_target_pid == trace_perf_samples);
+
+  ASSERT_OK(rows.Status());
+}
+
+// TODO (edgararriagag): There is currently a bug in the timeline where some
```

Comment:
Just to double check, what pids are the slices using? If they are outside of the target process, they're assigned a new pid. If that's the case, its not a bug with the timeline - it's working as expected. It would actually require new functionality.